File: aula 2/aula3/Person.java
class Personagem {

    private String nome;
    private int vida;
    private int nivel;

    public Personagem() {
        this.nome = "Jane";
        this.vida = 100;
        this.nivel = 1;
    }

    public Personagem(String nome, int vida, int nivel) {
        setNome(nome);
        setVida(vida);
        setNivel(nivel);
    }

    public String getNome() {
        return nome;
    }

    public void setNome(String nome) {
        if (nome != null && !nome.trim().isEmpty()) {
            this.nome = nome;
        } else {
            System.out.println("Erro: o nome não pode ser vazio!");
        }
    }

    public int getVida() {
        return vida;
    }

    public void setVida(int vida) {
        if (vida >= 0 && vida <= 100) {
            this.vida = vida;
        } else {
            System.out.println("Erro: a vida deve estar entre 0 e 100!");
        }
    }

    public int getNivel() {
        return nivel;
    }

    public void setNivel(int nivel) {
        if (nivel >= 1) {
            this.nivel = nivel;
        } else {
            System.out.println("Erro: o nível deve ser no mínimo 1!");
        }
    }

    public void atacar() {
        System.out.println(nome + " realizou um ataque padrão de 10 de dano!");
    }

    public void atacar(int dano) {
        System.out.println(nome + " realizou um ataque de " + dano + " de dano!");
    }
}
public class Main {

    public static void main(String[] args) {

        Personagem personagem1 = new Personagem();

        Personagem personagem2 = new Personagem("Angela", 80, 5);

        System.out.println("-Primeiro personagem-");

        System.out.println("Nome: " + personagem1.getNome());
        System.out.println("Vida: " + personagem1.getVida());
        System.out.println("Nível: " + personagem1.getNivel());

        System.out.println();

        System.out.println("-Segundo personagem-");

        System.out.println("Nome: " + personagem2.getNome());
        System.out.println("Vida: " + personagem2.getVida());
        System.out.println("Nível: " + personagem2.getNivel());

        System.out.println();

        personagem1.atacar();

        personagem2.atacar(50);
    }
}

File: aula 2/aula3/person.py
class Personagem:

    def __init__(self, nome="Jane", vida=100, nivel=1):
        self.nome = nome
        self.vida = vida
        self.nivel = nivel

    @property
    def nome(self):
        return self._nome

    @nome.setter
    def nome(self, nome):
        if nome is not None and nome.strip() != "":
            self._nome = nome
        else:
            print("Erro: o nome não pode ser vazio!")

    @property
    def vida(self):
        return self._vida

    @vida.setter
    def vida(self, vida):
        if vida >= 0 and vida <= 100:
            self._vida = vida
        else:
            print("Erro: a vida deve estar entre 0 e 100!")

    @property
    def nivel(self):
        return self._nivel

    @nivel.setter
    def nivel(self, nivel):
        if nivel >= 1:
            self._nivel = nivel
        else:
            print("Erro: o nível deve ser no mínimo 1!")

    def atacar(self, dano=10):
        print(self.nome + " realizou um ataque de " + str(dano) + " de dano!")


personagem1 = Personagem()

personagem2 = Personagem("Angela", 80, 5)

print("-Primeira personagem-")

print("Nome:", personagem1.nome)
print("Vida:", personagem1.vida)
print("Nível:", personagem1.nivel)

print()

print("-Segunda personagem-")

print("Nome:", personagem2.nome)
print("Vida:", personagem2.vida)
print("Nível:", personagem2.nivel)

print()

personagem1.atacar()

personagem2.atacar(50)
